src/rgw/rgw_http_client.h

```diff
@@ -91,6 +91,52 @@ class RGWHTTPClient
   int get_req_retcode();
 };
 
+
+class RGWHTTPHeadersCollector : public RGWHTTPClient {
+public:
+  /* Case insensitive comparator for containers carrying HTTP headers. */
+  struct CILess : public std::binary_function<std::string, std::string, bool> {
```

Comment:
It looks that `CILess` can be dropped as `rgw_string.h` has `lstr_nocase` we can reuse. :-) Will switch soon.